plugins/destination/s3/client/write.go

```diff
@@ -127,3 +128,21 @@ func sanitizeJSONKeysForObject(data any) any {
 		return data
 	}
 }
+
+func (c *Client) contentType() string {
+	if c.spec.ContentType != "" {
+		return c.spec.ContentType
+	}
+	switch c.spec.Format {
```

Comment:
If Compression is not empty what value should we set?

Reply:
if it's `gzip` it should be `application/gzip`. Or we can just leave it alone.

Reply:
I am going to leave it as it is because according to [MDN Web Docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding):

`Note that the original media/content type is specified in the [Content-Type](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type) header, and that the Content-Encoding applies to the representation, or "coded form", of the data. If the original media is encoded in some way (e.g. a zip file) then this information would not be included in the Content-Encoding header.`

Reply:
Can we set `ContentEncoding`?

Reply:
@disq pointed out that `Content-Encoding` describes the data as it is in transport not necessarily how it is on disc. In our case the data is actually a `gzip` so the `ContentType` should reflect that

[5b73a49](https://github.com/cloudquery/cloudquery/pull/17680/commits/5b73a497f4e4320fa987ba190e625eec0d04b3b8)